### DISCUSSION TITLE
Unexpected behaviour of controller

### DISCUSSION THREAD

**Discussion by AmoghBhagwat** [cyberbotics/wrestling#115] [General]
Hi @omichel
It seems there is some unpredictability in the behaviour of the controllers. In some runs the controller runs just fine, and in others it does not load in time and behaves erratically. Please have a look at the following videos, all triggered today.
The unexpected behaviour is seen [here](https://iitk-my.sharepoint.com/:v:/g/personal/bhagwata22_iitk_ac_in/Eb22UgyWKEFCm04gmBXXMWsB7-CyYoytFaPzLsG6xvq_Rg?e=MPAzfA). Our controller does not start for the first 12 seconds and so does not behave as it was designed to. [Here](https://iitk-my.sharepoint.com/:v:/g/personal/bhagwata22_iitk_ac_in/EU0gxK49iQNKh1PZMGcBpH8BRtrwQXxxoqlBj-zOn9m-8Q?e=n6C8wP) is what is supposed to happen. 
Also when triggered against the exact same opponent in the first case, it behaves as it supposed to. I request you to please look into this issue, or if it occurs in the finals to please restart the game. Thank you

**Comment by omichel** [comment]
I see. I will try to make my best to cancel games with such a strange behavior for all competitors.

**Reply by AmoghBhagwat** [reply]
Thanks a lot

**Reply by AmoghBhagwat** [reply]
Hi @omichel
It seems the issue happened again in our first match today (Team Humanoid, IITK vs T), although it didn't happen in the subsequent matches. My guess is that the docker container takes time to load initially and works perfectly for the rest of the matches. If you can add some delay after loading maybe the issue can be fixed.

**Reply by omichel** [reply]
I am afraid that adding a delay at this point of the competition may create other problems... But I will check the behavior of your controller in subsequent games to ensure the problem doesn't penalize you.
